src/composables/transactionList.ts

```diff
@@ -0,0 +1,272 @@
+    try {
+      const response = await fetchJson(
+        `${nodeNetworkId.value}/cache/events/?address=${address}&event=TipWithdrawn${recent ? '&limit=5' : ''}`,
+      );
+      if (response.message) return [];
+      const tipWithdrawnTransactions = (response as any[]).map(({
+        amount,
+        contract,
+        height,
+        data: { tx },
+        ...t
+      }) => ({
+        tx: {
+          ...tx,
+          address,
+          amount,
+          contractId: contract,
+          type: Tag[Tag.ContractCallTx],
+        },
+        ...t,
+        microTime: new Date(t.createdAt).getTime(),
+        blockHeight: height,
+        claim: true,
+      }));
+
+      return tipWithdrawnTransactions as ITransaction[];
+    } catch (eror) {
+      return [];
+    }
+  }
+
+  function removePendingTransactionByAccount(address: Encoded.AccountAddress, hash: string) {
+    const pendingTransactionForAccount: ITransaction[] = (
+      transactions.value[address]?.pending?.[nodeNetworkId.value!] || []
+    );
+
+    if (pendingTransactionForAccount.length) {
+      transactions.value[address].pending[nodeNetworkId.value!] = (
+        pendingTransactionForAccount.filter((transaction) => transaction.hash !== hash)
+      );
+    }
+  }
+
+  async function fetchTransactions(param: IFetchTransactionsParams): Promise<ITransaction[]> {
+    const transactionState = getOrInitTransactionStateForAccount(param.address);
+
+    if (transactionState.nextPageUrl === null && !param.recent) {
+      return [];
+    }
+    const { getMiddleware } = useMiddleware({ store });
+
+    const [
+      pending,
+      normal,
+      tip,
+    ] = await Promise.all([
+      fetchTransactiosFromMiddleware(transactionState, param.address, param.recent, param.limit),
+      fetchPendingTransaction(param.address),
+      fetchTipWithdrawnTransactions(param.address, param.recent),
+    ]);
+
+    const lastPendingTransaction = pending?.[pending.length - 1];
+    if (lastPendingTransaction?.type === AEX9_TRANSFER_EVENT) {
+      const middleware = await getMiddleware();
+      pending[pending.length - 1] = await middleware.getTx(lastPendingTransaction.payload.txHash);
+    }
+
+    let preparedTransactions = [
+      ...pending,
+      ...normal,
+    ]
+      .filter(({ type }) => !type?.startsWith('Internal'))
+      .map((transaction) => ({
+        ...(transaction.payload || transaction),
+        transactionOwner: param.address,
+        ...(transaction.type === AEX9_TRANSFER_EVENT
+          ? {
+            tx: {
+              ...transaction.payload,
+              callerId: transaction.payload.senderId,
+              type: 'ContractCallTx',
+            },
+            hash: transaction.payload.txHash,
+            incomplete: true,
+          } as ITransaction
+          : {}),
+      }));
+
+    preparedTransactions = uniqBy(preparedTransactions.reverse(), 'hash').reverse();
```

Comment:
Reverse is being called twice

Reply:
yes, and in this case it is a correct. Data should be reverse for correct making uniq, after that we should reverse again to have proper order. This code is copy from old store action.

Reply:
Okay.